utils/arch.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// UnderlyingArch detects the underlying architecture(amd64/arm64) of the system
+func UnderlyingArch() string {
+	name := "uname"
+	arg0 := "-m"
+
+	cmd := exec.Command(name, arg0)
+	stdout, err := cmd.Output()
+	if err != nil {
+		fmt.Errorf("Error finding the underlying architecture: %s", err.Error())
```

Comment:
@pskrbasu this creates an error object. doesn't actually print it. Return it. Look above